draco/schema.py

```diff
@@ -44,50 +45,52 @@ class Schema(TypedDict):
     field: list[FieldProps]
 
 
-def dtype_to_field_type(ty: DtypeObj) -> FieldType:
+def dtype_to_field_type(ty: nw.dtypes.DType) -> FieldType:
     """Simple converter that translates Pandas column types to data types for Draco."""
-    if ty in ["float64", "int64"]:
+    if ty.is_numeric():
         return "number"
-    elif ty in ["bool"]:
+    elif isinstance(ty, nw.dtypes.Boolean):
         return "boolean"
-    elif ty in ["object"]:
+    elif isinstance(ty, nw.dtypes.String):
         return "string"
-    elif ty in ["datetime64[ns]"]:
+    elif ty.is_temporal():
         return "datetime"
     else:
         raise ValueError(f"unsupported type {ty}")
```

Comment:
The primary benefit of this PR is that mapping `df` field types to our internal field types gets significantly higher coverage through a nice API without being forced to maintain the mapping on our side.